pkg/controller/reconciler/flux_gitrepository_predicate.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2024 The Radius Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1"
+)
+
+// GitRepositoryRevisionChangePredicate triggers an update event
+// when a GitRepository revision changes.
+type GitRepositoryRevisionChangePredicate struct {
+	predicate.Funcs
+}
+
+func (GitRepositoryRevisionChangePredicate) Create(e event.CreateEvent) bool {
```

Comment:
Would a pointer receiver make more sense to prevent unnecessary copying?
```suggestion
func (*GitRepositoryRevisionChangePredicate) Create(e event.CreateEvent) bool {
```